### PR TITLE
[SPARK-35206][TESTS][SQL] Extract common used get project path into a function in SparkFunctionSuite

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Add a common functions `getWorkspaceFilePath` (which prefixed with spark home) to `SparkFunctionSuite`, and applies these the function to where they're extracted from.

### Why are the changes needed?

Spark sql has test suites to read resources when running tests. The way of getting the path of resources is commonly used in different suites. We can extract them into a function to ease the code maintenance.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass existing tests.

### CODE REVIEW

**Comment by Ngone51** [comment]
cc @cloud-fan @yhuai please take a look, thanks!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala [hunk 1]
```diff
@@ -27,11 +27,11 @@ import org.apache.spark.SparkFunSuite
 import org.apache.spark.sql.catalyst.plans.SQLHelper
 import org.apache.spark.sql.catalyst.util.fileToString
 
-trait SQLKeywordUtils extends SQLHelper {
+trait SQLKeywordUtils extends SparkFunSuite with SQLHelper {
 
   val sqlSyntaxDefs = {
     val sqlBasePath = {
-      java.nio.file.Paths.get(sparkHome, "sql", "catalyst", "src", "main", "antlr4", "org",
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala:34]: can we remove the lazy val `sparkHome`?

Comment (Ngone51) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala:34]: It's still used in `IntegratedUDFTestUtils`.

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala [hunk 2]
```diff
@@ -119,6 +121,27 @@ abstract class SparkFunSuite
...
+   * Get a Path relative to the project containing the test.
+   * It is assumed that tests are executed from the project's root directory,
+   * so the returned path is simply a relative path composed of the given elements.
+   */
+  protected final def getProjectFilePath(first: String, more: String*): Path = {
+    java.nio.file.Paths.get(first, more: _*)
+  }
+
+  /**
+   * Get a Path relative to the root project containing the test.
```
Comment (szeiger) [inline core/src/test/scala/org/apache/spark/SparkFunSuite.scala:134]: I'd remove "containing the test" here to make it clearer.

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala [hunk 3]
```diff
@@ -119,6 +121,26 @@ abstract class SparkFunSuite
...
+   * path composed of the given elements.
+   */
+  protected final def getProjectFilePath(first: String, more: String*): Path = {
+    java.nio.file.Paths.get(first, more: _*)
+  }
+
+  /**
+   * Get a Path relative to the root project. It is assumed that a spark home is set.
+   */
+  protected final def getWorkspaceFilePath(first: String, more: String*): Path = {
```
[the remark this answers is not in the document]
Comment (szeiger) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:136]: It's not relative vs absolute, but relative to the subproject vs relative to the root project in sbt terms. I'm not sure what the clearest terms for Maven would be. Ideally we would only have one of these methods that is used by all tests.

[the remark this answers is not in the document]
Comment (cloud-fan) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:136]: makes sense. Can we remove `getProjectFilePath`? We can change the caller side to call `getWorkspaceFilePath` with the subproject path, e.g. `getWorkspaceFilePath("sql", "core", the original path)`

[the remark this answers is not in the document]
Comment (cloud-fan) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:136]: @Ngone51 can you help to make this change?

[the remark this answers is not in the document]
Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:127]: Updated, please take another look @cloud-fan @szeiger

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala [hunk 4]
```diff
@@ -19,11 +19,13 @@ package org.apache.spark
 
 // scalastyle:off
 import java.io.File
+import java.nio.file.Path
 import java.util.{Locale, TimeZone}
 
 import org.apache.log4j.spi.LoggingEvent
```
Comment (cloud-fan) [inline core/src/test/scala/org/apache/spark/SparkFunSuite.scala:25]: seems the import order is wrong. This should appear after `scala.annotation.tailrec`

**Comment by cloud-fan** [comment]
looks good, can you update the PR description?

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala (hunk 4 above)
Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala]: Not sure why the scala style check has been disabled for imports. I have re-organized them.

**Comment by Ngone51** [comment]
> looks good, can you update the PR description?

Updated, thanks!

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**